resources/puppetlabs/puppetdb/pql/pql-grammar-experimental.ebnf

```diff
@@ -0,0 +1,139 @@
+(* PQL experimental EBNF specification
+
+   This file is the formal specification used by our parsing library that defines
+   the PQL grammar for translation to the AST query language.
+
+   This file is used by Mark Engelberg's 'Instaparse' parser in clojure, and
+   follows its own form of EBNF, so is not strictly 'EBNF' but an augmentation. *)
+
+(* TODO: the following items are just not supported in the syntax yet:
+
+   - group_by
+   - aggregate functions
+   - ~>
+   - in <array> *)
+
+(* TODO: the following features are awaiting PDB AST changes first, but
+   generally have support within PDB in one form or another:
+
+   - order by
+   - limit, offset
+   - include_total *)
```

Comment:
You mean aggregate-event-counts? Why and in what way? If you mean we should provide mechanisms that provide aggregate like a capabilities I'm all ears, but making them work in PQL as they do today seems like folly.